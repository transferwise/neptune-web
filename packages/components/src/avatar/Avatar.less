@import (reference) '@transferwise/neptune-css/src/variables/neptune-tokens.less';

@avatar-diameter-lg: 72px;
@avatar-diameter-md: 48px;
@avatar-diameter-sm: 24px;

.tw-avatar {
  position: relative;
  border-radius: 50%;
  user-select: none;
  box-sizing: border-box;
  cursor: default;

  & &__content {
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    overflow: hidden;
    background-color: var(--color-background-neutral);
    height: 100%;
    width: 100%;
    max-height: 100%;
    max-width: 100%;
  }

  &--outlined {
    border: 1px solid var(--color-control-accent);
    cursor: pointer;

    &:hover {
      border-color: var(--color-control-accent-hover);
    }
  }

  &--dark &__content {
    background: @color-base-navy-mid;
    color: @color-text-control;
  }

  &--thumbnail {
    & * {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }

  &--icon {
    & * {
      max-height: 100%;
      max-width: 100%;
    }
  }

  &.tw-avatar--initials,
  &.tw-avatar--icon {
    color: @color-text-primary;
  }

  &.tw-avatar--initials,
  &.tw-avatar--emoji {
    // There is a problem how emojis & text vertically aligned between Windows and Mac. Setting font family to Arial seems to resolve it.
    font-family: Arial, sans-serif;
  }

  &--sm {
    .avatar-dimension(@avatar-diameter-sm, 1px);
  }

  &--md {
    .avatar-dimension(@avatar-diameter-md, 2px);
  }

  &--lg {
    .avatar-dimension(@avatar-diameter-lg, 3px);
  }

  img {
    width: 100%;
    height: 100%;
  }
}

.avatar-dimension(@avatar-size, @mask-size) {
  min-width: @avatar-size;
  width: @avatar-size;
  height: @avatar-size;

  &.tw-avatar--emoji,
  &.tw-avatar--icon {
    font-size: @avatar-size * 0.5;
  }

  &.tw-avatar--initials {
    font-size: @avatar-size * 0.35;
    font-weight: bold;
  }

  &.tw-avatar--outlined {
    padding: @mask-size;

    &.tw-avatar--thumbnail::after {
      top: @mask-size;
      left: @mask-size;
      right: @mask-size;
      bottom: @mask-size;
    }
  }

  &.tw-avatar--thumbnail::after {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.08);
    content: '';
    border-radius: 50%;
  }

  &.tw-avatar--dark.tw-avatar--thumbnail::after {
    border-color: rgba(255, 255, 255, 0.08);
  }
}
