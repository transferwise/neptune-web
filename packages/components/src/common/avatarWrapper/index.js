import AvatarWrapper from './AvatarWrapper';

export default AvatarWrapper;
