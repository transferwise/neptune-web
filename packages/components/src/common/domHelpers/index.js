export function isTouchDevice() {
  return !!window?.matchMedia('(pointer: coarse)')?.matches;
}

export {
  addClickClassToDocumentOnIos,
  removeClickClassFromDocumentOnIos,
} from './documentIosClick';
