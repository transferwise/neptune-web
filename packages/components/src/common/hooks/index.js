export { default as useHasIntersected } from './useHasIntersected';
export { default as useClientWidth } from './useClientWidth';
