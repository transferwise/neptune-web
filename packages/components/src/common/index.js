export { default as HistoryNavigator } from './historyNavigator';
export { addNoScrollBodyClass, removeNoScrollBodyClass } from './DOMOperations';
export { Theme } from './theme';
// Props
export { Breakpoint } from './propsValues/breakpoint';
export { Size } from './propsValues/size';
export { Type } from './propsValues/type';
export { DateMode } from './propsValues/dateMode';
export { MonthFormat } from './propsValues/monthFormat';
// Position and Placement are duplicated and needs to be aligned with a breaking change
export { Position } from './propsValues/position';
export { Layout } from './propsValues/layouts';
export { FormControlType } from './propsValues/formControlType';
export { Status } from './propsValues/status';
export { Sentiment } from './propsValues/sentiment';
export { ProfileType } from './propsValues/profileType';
export { Key } from './key';
