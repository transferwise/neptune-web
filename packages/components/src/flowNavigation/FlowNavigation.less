@import (reference) '@transferwise/neptune-css/src/less/addons/_spacing-utilities.less';

@flag: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPCFET0NUWVBFIHN2ZyBQVUJMSUMgIi0vL1czQy8vRFREIFNWRyAxLjEvL0VOIiAiaHR0cDovL3d3dy53My5vcmcvR3JhcGhpY3MvU1ZHLzEuMS9EVEQvc3ZnMTEuZHRkIj4KPHN2ZyB2ZXJzaW9uPSIxLjEiIGlkPSJ0cmFuc2Zlcndpc2UtZmxhZyIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IgoJIHk9IjBweCIgdmlld0JveD0iMCAwIDQ4IDUyIiBlbmFibGUtYmFja2dyb3VuZD0ibmV3IDAgMCA0OCA1MiIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI+CjxwYXRoIGZpbGw9IiMwMEM2RkYiIGQ9Ik01LjgsMGw4LjIsMTMuNkwwLjEsMjcuMmgyNC4xbDIuNi01LjRIMTNsOC4yLTguM2wtNS04aDIyLjZMMTkuMiw1Mmg2LjdsMjItNTJINS44eiIvPgo8L3N2Zz4=);

.np-flow-navigation {
  width: 100%;
  // This prevents jumping when border disappears.
  min-height: 97px;
  &--border-bottom {
    // This is the natural height of FlowNavigation when the Avatar is displayed.
    min-height: 96px;
    border-bottom: 1px solid var(--color-border-neutral);
  }

  &__content {
    width: 100%;
    min-width: 320px;
    max-width: 1164px;
  }

  &__stepper {
    .p-b-0;
    .p-x-0;
  }

  &--xs-max {
    .np-flow-navigation__stepper {
      .tw-stepper-steps {
        display: none;
      }
      .progress {
        .m-b-0;
      }
    }
  }

  &--sm {
    .np-flow-navigation__stepper {
      .p-x-3;
      min-height: 56px;
    }
  }

  &--lg {
    .np-flow-navigation__stepper {
      max-width: 562px;
    }
  }

  .separator {
    .m-x-2;
    height: var(--size-32);
    width: 1px;
    min-width: 1px;
    background-color: var(--color-border-neutral);
  }

  &--hidden {
    visibility: hidden;
  }

  &--flag {
    width: var(--size-24);
    height: var(--size-32);

    background-repeat: no-repeat;
    background-position: 50% 50%;
    background-image: @flag;

    transition: transform 0.3s ease-in-out, opacity 0.3s ease-in-out;

    &__hidden {
      position: absolute;
      opacity: 0;
      transform: rotate(90deg);
      margin-top: -24px;
      z-index: -1;
    }
    &__display {
      opacity: 1;
      transform: rotate(0deg);
    }
  }

  &--xs-max {
    .np-animated-label {
      height: auto;
    }
  }
}
