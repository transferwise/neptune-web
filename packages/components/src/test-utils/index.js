import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from '..';
import { DEFAULT_LOCALE } from '../common/locale';
import en from '../../i18n/en.json';

/**
 * Custom `render` function which wraps passed elements in Provider component
 * Source: https://testing-library.com/docs/example-react-intl/#creating-a-custom-render-function
 */
function customRender(ui, { locale = DEFAULT_LOCALE, messages = en, ...renderOptions } = {}) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return <Provider i18n={{ locale, messages }}>{children}</Provider>;
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { userEvent };
export { customRender as render };
