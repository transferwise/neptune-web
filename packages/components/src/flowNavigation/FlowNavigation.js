import React, { useRef } from 'react';
import Types from 'prop-types';
import classNames from 'classnames';
import Header from '../header';
import Stepper from '../stepper';
import AnimatedLabel from './animatedLabel';
import BackButton from './backButton';
import CloseButton from '../common/closeButton';

import { Theme, Breakpoint } from '../common';
import { useClientWidth } from '../common/hooks';

import './FlowNavigation.css';

// Size switches on parent container which may or may not have the same size as the window.
const containerBreakpoints = {
  'np-flow-navigation--sm': Breakpoint.SMALL,
  'np-flow-navigation--lg': Breakpoint.LARGE,
};

const FlowNavigation = ({ activeStep, avatar, logo, onClose, onGoBack, done, steps }) => {
  const ref = useRef(null);

  const [clientWidth] = useClientWidth({ ref });
  const closeButton = onClose && <CloseButton onClick={onClose} />;
  const isSmall = clientWidth < Breakpoint.SMALL;

  const newAvatar = done ? null : avatar;

  const getLeftContentSmall = () => {
    const displayGoBack = onGoBack && activeStep > 0;
    return (
      <>
        {displayGoBack && (
          <BackButton
            label={
              <AnimatedLabel
                className="m-x-1"
                labels={steps.map((step) => step.label)}
                activeLabel={activeStep - 1}
              />
            }
            onClick={onGoBack}
          />
        )}
        <div
          className={classNames('np-flow-navigation--flag', {
            'np-flow-navigation--flag__hidden': displayGoBack,
            'np-flow-navigation--flag__display': !displayGoBack,
          })}
        />
      </>
    );
  };

  return (
    <div
      ref={ref}
      className={classNames(
        'np-flow-navigation d-flex align-items-center justify-content-center p-y-3',
        { 'np-flow-navigation--border-bottom': !done },
      )}
    >
      <Header
        className={classNames(
          'np-flow-navigation__content p-x-3',
          {
            'np-flow-navigation--hidden': !clientWidth,
            'np-flow-navigation--xs-max': isSmall,
          },
          Object.keys(containerBreakpoints).filter(
            (key) => clientWidth >= containerBreakpoints[key],
          ),
        )}
        leftContent={isSmall ? getLeftContentSmall() : logo}
        rightContent={
          <>
            {newAvatar}
            {newAvatar && closeButton && <span className="separator" />}
            {closeButton}
          </>
        }
        bottomContent={
          !done && (
            <Stepper
              activeStep={activeStep}
              steps={steps}
              className={classNames('np-flow-navigation__stepper')}
            />
          )
        }
        layout={clientWidth < Breakpoint.LARGE ? Header.Layout.VERTICAL : Header.Layout.HORIZONTAL}
      />
    </div>
  );
};

FlowNavigation.Theme = Theme;

FlowNavigation.defaultProps = {
  activeStep: 0,
  avatar: undefined,
  onGoBack: undefined,
  onClose: undefined,
  done: false,
};

FlowNavigation.propTypes = {
  activeStep: Types.number,
  /** An Avatar */
  avatar: Types.element,
  logo: Types.node.isRequired,
  /** Called when the close button is clicked. If not provided the close button won't show */
  onClose: Types.func,
  /** Called when the back button is clicked. If not provided the back button won't show. The back button only shows on small screens */
  onGoBack: Types.func,
  done: Types.bool,
  /** Steps to be displayed in stepper. If you don't need the stepper, please use OverlayHeader instead */
  steps: Types.arrayOf(
    Types.shape({
      label: Types.node.isRequired,
      onClick: Types.func,
      hoverLabel: Types.node,
    }),
  ).isRequired,
};

export default FlowNavigation;
