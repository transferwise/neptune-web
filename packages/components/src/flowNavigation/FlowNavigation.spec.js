import React from 'react';
import '@testing-library/jest-dom';

import { render, screen } from '../test-utils';

import FlowNavigation from '.';
import { Breakpoint } from '../common';

import Avatar from '../avatar';

jest.mock('lodash.throttle', () => jest.fn((fn) => fn));

jest.mock('./animatedLabel', () => {
  // eslint-disable-next-line react/prop-types
  return ({ className, activeLabel }) => (
    <div className={className} data-testid={`activeLabel-${activeLabel}`}>
      AnimatedLabel
    </div>
  );
});
jest.mock('./backButton', () => {
  // eslint-disable-next-line react/prop-types
  return ({ className, label }) => <div className={className}>BackButton{label}</div>;
});

describe('FlowNavigation', () => {
  const originalClientWidth = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'clientWidth');
  const resetClientWidth = (width) => {
    Object.defineProperty(HTMLElement.prototype, 'clientWidth', {
      configurable: true,
      value: width,
    });
  };
  beforeEach(() => {
    resetClientWidth(Breakpoint.LARGE + 1);
  });

  afterAll(() => {
    Object.defineProperty(HTMLElement.prototype, 'clientWidth', originalClientWidth);
  });

  const props = {
    avatar: (
      <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.MEDIUM}>
        TM
      </Avatar>
    ),
    logo: <img alt="logo" src="logo.svg" width="138" height="24" />,
    onClose: jest.fn(),
    steps: [
      {
        label: 'label-0',
      },
      { label: 'label-1' },
      { label: 'label-2' },
    ],
    activeStep: 0,
  };

  it(`renders as expected`, () => {
    expect(render(<FlowNavigation {...props} />).container).toMatchSnapshot();
  });

  it(`renders full Logo not visible on mobile`, () => {
    render(<FlowNavigation {...props} />);
    expect(logoFull()).toBeInTheDocument();
  });

  it(`renders separator if avatar and onClose are provided`, () => {
    const { container } = render(<FlowNavigation {...props} />);

    expect(container.querySelector('.separator')).toBeInTheDocument();
  });

  it(`doesn't render separator if avatar or onClose are not provided `, () => {
    const { container, rerender } = render(<FlowNavigation {...props} onClose={null} />);

    expect(container.querySelector('.separator')).not.toBeInTheDocument();

    rerender(<FlowNavigation {...props} avatar={null} />);

    expect(container.querySelector('.separator')).not.toBeInTheDocument();
  });

  it(`doesn't render separator if done is true `, () => {
    const { container, rerender } = render(<FlowNavigation {...props} onClose={null} />);

    expect(container.querySelector('.separator')).not.toBeInTheDocument();

    rerender(<FlowNavigation {...props} avatar={null} done />);

    expect(container.querySelector('.separator')).not.toBeInTheDocument();
  });

  it(`renders border based on done `, () => {
    const { container, rerender } = render(<FlowNavigation {...props} onClose={null} />);

    expect(container.querySelector('.np-flow-navigation--border-bottom')).toBeInTheDocument();

    rerender(<FlowNavigation {...props} avatar={null} done />);

    expect(container.querySelector('.np-flow-navigation--border-bottom')).not.toBeInTheDocument();
  });

  it(`hides stepper when done is true`, () => {
    const { container, rerender } = render(<FlowNavigation {...props} onClose={null} />);

    expect(container.querySelector('.np-flow-navigation__stepper')).toBeInTheDocument();

    rerender(<FlowNavigation {...props} avatar={null} done />);

    expect(container.querySelector('.np-flow-navigation__stepper')).not.toBeInTheDocument();
  });

  it(`renders xs-max class`, () => {
    resetClientWidth(Breakpoint.SMALL - 1);
    const { container } = render(<FlowNavigation {...props} onClose={null} />);

    expect(container.querySelector('.np-flow-navigation--xs-max')).toBeInTheDocument();
  });

  it(`renders sm class`, () => {
    resetClientWidth(Breakpoint.SMALL);
    const { container } = render(<FlowNavigation {...props} onClose={null} />);

    expect(container.querySelector('.np-flow-navigation--sm')).toBeInTheDocument();
  });

  it(`renders lg class`, () => {
    resetClientWidth(Breakpoint.LARGE);
    const { container } = render(<FlowNavigation {...props} onClose={null} />);

    expect(container.querySelector('.np-flow-navigation--lg')).toBeInTheDocument();
  });

  describe('on mobile', () => {
    beforeEach(() => {
      resetClientWidth(Breakpoint.SMALL - 1);
    });

    it('renders as expected', () => {
      expect(render(<FlowNavigation {...props} />).container).toMatchSnapshot();
    });

    it(`renders full Logo not visible on mobile`, () => {
      render(<FlowNavigation {...props} />);
      expect(screen.queryByAltText(`logo`)).not.toBeInTheDocument();
    });

    it(`renders flag if activeStep <= 0 onGoBack or is not provided`, () => {
      const { container, rerender } = render(
        <FlowNavigation {...props} activeStep={0} onGoBack={undefined} />,
      );

      const flag = container.querySelector('.np-flow-navigation--flag');

      expect(flag).toBeInTheDocument();
      expect(flag).toHaveClass('np-flow-navigation--flag__display');

      rerender(<FlowNavigation {...props} activeStep={1} onGoBack={undefined} />);

      expect(flag).toHaveClass('np-flow-navigation--flag__display');

      rerender(<FlowNavigation {...props} activeStep={0} onGoBack={jest.fn()} />);

      expect(flag).toHaveClass('np-flow-navigation--flag__display');

      rerender(<FlowNavigation {...props} onGoBack={jest.fn()} activeStep={1} />);

      expect(flag).toHaveClass('np-flow-navigation--flag__hidden');
    });

    it('renders BackButton with AnimatedLabel if onGoBack is provided and activeStep > 0', () => {
      const { rerender } = render(<FlowNavigation {...props} onGoBack={jest.fn()} />);

      expect(screen.queryByText('BackButton')).toBeNull();

      rerender(<FlowNavigation {...props} activeStep={1} />);
      expect(screen.queryByText('BackButton')).toBeNull();

      rerender(<FlowNavigation {...props} onGoBack={jest.fn()} activeStep={1} />);

      expect(screen.getByText('BackButton')).toBeInTheDocument();
      expect(screen.getByText('AnimatedLabel')).toBeInTheDocument();
    });

    it('renders correct AnimatedLabel', () => {
      const { rerender } = render(
        <FlowNavigation {...props} onGoBack={jest.fn()} activeStep={1} />,
      );

      expect(screen.getByTestId('activeLabel-0')).toBeInTheDocument();

      rerender(<FlowNavigation {...props} onGoBack={jest.fn()} activeStep={2} />);

      expect(screen.getByTestId('activeLabel-1')).toBeInTheDocument();
    });
  });
  const logoFull = () => screen.getByAltText(`logo`);
});
