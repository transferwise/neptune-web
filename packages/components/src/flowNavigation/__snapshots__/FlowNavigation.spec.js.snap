// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowNavigation on mobile renders as expected 1`] = `
<div>
  <div
    class="np-flow-navigation d-flex align-items-center justify-content-center p-y-3 np-flow-navigation--border-bottom"
  >
    <div
      class="np-header d-flex flex-wrap np-flow-navigation__content p-x-3 np-flow-navigation--xs-max"
    >
      <div
        class="align-items-center d-flex flex__item--8"
      >
        <div>
          BackButton
          <div
            class="m-x-1"
            data-testid="activeLabel-0"
          >
            AnimatedLabel
          </div>
        </div>
        <div
          class="np-flow-navigation--flag np-flow-navigation--flag__hidden"
        />
      </div>
      <div
        class="align-items-center d-flex justify-content-end flex__item--4 "
      >
        <div
          class="tw-avatar tw-avatar--md tw-avatar--initials tw-avatar--light"
        >
          <div
            class="tw-avatar__content"
          >
            TM
          </div>
        </div>
        <span
          class="separator"
        />
        <button
          aria-label="Close"
          class="np-close-button close btn-link text-no-decoration np-close-button--large"
          type="button"
        >
          <span
            aria-hidden="true"
            class="tw-icon tw-icon-cross "
            role="presentation"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="24"
              width="24"
            >
              <path
                d="M12 13.414l7.293 7.293 1.414-1.414L13.414 12l7.293-7.293-1.414-1.414L12 10.586 4.707 3.293 3.293 4.707 10.586 12l-7.293 7.293 1.414 1.414L12 13.414z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="align-items-center d-flex justify-content-center flex__item--12"
      >
        <div
          class="tw-stepper np-flow-navigation__stepper"
        >
          <div
            class="progress"
          >
            <div
              class="progress-bar-filler"
              style="width: 0%;"
            />
            <div
              class="progress-bar-ending"
              style="width: 50%;"
            />
          </div>
          <ul
            class="tw-stepper-steps p-t-1 m-b-0"
          >
            <li
              class="
          hidden-xs
          tw-stepper__step
          
          
        "
              style="left: 0%;"
            >
              <button
                class="btn-unstyled tw-stepper__step-label"
                disabled=""
                type="button"
              >
                <small>
                  label-0
                </small>
              </button>
            </li>
            <li
              class="
          hidden-xs
          tw-stepper__step
          tw-stepper__step--active
          
        "
              style="left: 50%;"
            >
              <button
                class="btn-unstyled tw-stepper__step-label"
                disabled=""
                type="button"
              >
                <small>
                  label-1
                </small>
              </button>
            </li>
            <li
              class="
          hidden-xs
          tw-stepper__step
          
          
        "
              style="left: 100%;"
            >
              <button
                class="btn-unstyled tw-stepper__step-label"
                disabled=""
                type="button"
              >
                <small>
                  label-2
                </small>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FlowNavigation renders as expected 1`] = `
<div>
  <div
    class="np-flow-navigation d-flex align-items-center justify-content-center p-y-3 np-flow-navigation--border-bottom"
  >
    <div
      class="np-header d-flex flex-wrap np-flow-navigation__content p-x-3 np-flow-navigation--sm np-flow-navigation--lg"
    >
      <div
        class="align-items-center d-flex"
      >
        <img
          alt="logo"
          height="24"
          src="logo.svg"
          width="138"
        />
      </div>
      <div
        class="align-items-center d-flex justify-content-end order-2"
      >
        <div
          class="tw-avatar tw-avatar--md tw-avatar--initials tw-avatar--light"
        >
          <div
            class="tw-avatar__content"
          >
            TM
          </div>
        </div>
        <span
          class="separator"
        />
        <button
          aria-label="Close"
          class="np-close-button close btn-link text-no-decoration np-close-button--large"
          type="button"
        >
          <span
            aria-hidden="true"
            class="tw-icon tw-icon-cross "
            role="presentation"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="24"
              width="24"
            >
              <path
                d="M12 13.414l7.293 7.293 1.414-1.414L13.414 12l7.293-7.293-1.414-1.414L12 10.586 4.707 3.293 3.293 4.707 10.586 12l-7.293 7.293 1.414 1.414L12 13.414z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="align-items-center d-flex justify-content-center order-1 flex-grow-1"
      >
        <div
          class="tw-stepper np-flow-navigation__stepper"
        >
          <div
            class="progress"
          >
            <div
              class="progress-bar-filler"
              style="width: 0%;"
            />
            <div
              class="progress-bar-ending"
              style="width: 0%;"
            />
          </div>
          <ul
            class="tw-stepper-steps p-t-1 m-b-0"
          >
            <li
              class="
          hidden-xs
          tw-stepper__step
          tw-stepper__step--active
          
        "
              style="left: 0%;"
            >
              <button
                class="btn-unstyled tw-stepper__step-label"
                disabled=""
                type="button"
              >
                <small>
                  label-0
                </small>
              </button>
            </li>
            <li
              class="
          hidden-xs
          tw-stepper__step
          
          
        "
              style="left: 50%;"
            >
              <button
                class="btn-unstyled tw-stepper__step-label"
                disabled=""
                type="button"
              >
                <small>
                  label-1
                </small>
              </button>
            </li>
            <li
              class="
          hidden-xs
          tw-stepper__step
          
          
        "
              style="left: 100%;"
            >
              <button
                class="btn-unstyled tw-stepper__step-label"
                disabled=""
                type="button"
              >
                <small>
                  label-2
                </small>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
